packages/flutter/test/widgets/framework_test.dart

```diff
@@ -2495,3 +2505,39 @@ class _NullElement extends Element {
   @override
   bool get debugDoingBuild => throw UnimplementedError();
 }
+
+class _DiagnosticsOverrideWidget extends LeafRenderObjectWidget {
```

Comment:
@chunhtai is there prior art for this in the framework where we subclass solely to assert the api?

Reply:
No, we don't usually lock an API like this.